server/models/userModel.js

```diff
@@ -0,0 +1,26 @@
+import mongoose from "mongoose";
+
+const userSchema = mongoose.Schema(
+    {
+        username: {
+            type: String,
+            required: true,
```

Comment:
will also probably need to be unique

We may also in the future need to index these fields for faster lookup